java/src/main/java/ai/onnxruntime/OnnxSparseTensor.java

```diff
@@ -0,0 +1,876 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+import java.util.Arrays;
+
+/**
+ * A Java object wrapping an OnnxSparseTensor.
+ *
+ * <p>Sparse tensors support a variety of formats, and the {@link #getValue} method returns a
+ * different static inner class representing each type.
+ */
+public final class OnnxSparseTensor extends OnnxTensorLike {
+  private final SparseTensorType sparseTensorType;
+
+  // Held to prevent deallocation while used in native code.
+  private final Buffer indices;
+  private final LongBuffer innerIndices;
+  private final Buffer data;
```

Comment:
Just FYI. Both data and the indices are kept in one single buffer. Values first and then aligned indices right after.